src/website/app/pages/Theming/groupedMineralTheme.js

```diff
@@ -1,54 +1,36 @@
 /* @flow */
-import * as groupedTokens from 'mineral-ui-tokens/groupedTokens';
+import categorizedJsTokens from '../../categorizedJsTokens';
 import { black, brand, gray, white } from 'mineral-ui-tokens';
 import { createColorRamp, themeFromTokens } from '../../../../library/themes';
 import { nonTokenVariables } from '../../../../library/themes/createTheme';
 
-import type { Ramp, Theme } from '../../../../library/themes/themeFromTokens';
-
-type Group = [string, Theme | ((themeRamp?: Ramp) => Theme)];
-type GroupedTheme = Array<Group>;
+const { boxShadow_focusInner, direction, fontSize_base } = nonTokenVariables;
 
 const grayRamp = createColorRamp(gray, 'color_gray_');
-const themeRamp = undefined;
+const themeRamp = createColorRamp(brand, 'color_theme_');
 
-const getThemeRamp = (themeRamp?: Ramp) => {
-  return themeRamp || createColorRamp(brand, 'color_theme_');
-};
+export default Object.keys(categorizedJsTokens)
+  .filter((category) => category !== 'palette')
+  .map((category) => {
+    let tokens = categorizedJsTokens[category];
 
-const groupedMineralTheme: GroupedTheme = [
-  [
-    'generics',
-    (themeRamp) => {
-      const group = {
-        ...themeFromTokens({ themeRamp, tokens: groupedTokens.generic }),
-        ...nonTokenVariables
-      };
-      return Object.keys(group)
-        .sort()
-        .reduce((acc, key) => {
-          acc[key] = group[key];
-          return acc;
-        }, {});
+    if (category === 'depth') {
+      tokens = { ...tokens, boxShadow_focusInner };
+    } else if (category === 'typography') {
+      tokens = { ...tokens, fontSize_base };
     }
-  ],
-  ['Headings', themeFromTokens({ themeRamp, tokens: groupedTokens.heading })],
-  [
-    'Icons',
-    (themeRamp) => themeFromTokens({ themeRamp, tokens: groupedTokens.icon })
-  ],
-  ['Inputs', themeFromTokens({ themeRamp, tokens: groupedTokens.input })],
-  ['Panels', themeFromTokens({ themeRamp, tokens: groupedTokens.panel })],
-  ['Wells', themeFromTokens({ themeRamp, tokens: groupedTokens.well })],
-  [
-    'Colors',
-    (themeRamp) => ({
-      ...getThemeRamp(themeRamp),
-      color_white: white,
-      ...grayRamp,
-      color_black: black
-    })
-  ]
-];
 
-export default groupedMineralTheme;
+    return [category, themeFromTokens({ tokens })];
+  })
+  .concat([
+    ['Other', { direction }],
+    [
+      'Colors',
+      {
+        ...themeRamp,
+        color_white: white,
+        ...grayRamp,
+        color_black: black
+      }
+    ]
+  ]);
```

Comment:
We may want to sort these to be alphabetical. Otherwise, all the non-targeted (previously, generic) tokens should come before the targeted ones, except for the Palette, which should remain last.